generator/src/main/java/com/linkedin/pegasus/generator/JavaDataTemplateGenerator.java

```diff
@@ -1218,6 +1443,110 @@ private void generateConstructorWithMap(JDefinedClass cls, JClass valueClass)
     argConstructor.body().invoke("putAll").arg(m);
   }
 
+  private void generateOnUnderlyingMapChanged(JDefinedClass cls, Map<String, JVar> fieldMap)
+  {
+    final JMethod method = cls.method(JMod.PRIVATE, Void.class, "onUnderlyingMapChanged");
+    final JVar keyParam = method.param(String.class, "key");
+    method.param(_objectClass, "value");
+    JSwitch keySwitch = method.body()._switch(keyParam);
+    fieldMap.forEach((key, field) -> {
+      JCase keyCase = keySwitch._case(JExpr.lit(key));
+      keyCase.body().assign(field, JExpr._null());
```

Comment:
What if we set the instance vars for primitive fields to the value if the value is not null? It would avoid a map lookup when the field is accessed later.
eg,
```
private Void onUnderlyingMapChanged(String key, Object value) {
switch (key) {
           // _owner_name is primitive, so we can initialize the field
            case "owner_name":
                _owner_name = value instanceOf String.class ? (String) value : null;
                return null;
            case "acls":
               // _acls is ACLArray so no need to wrap unless the field is accessed. 
                _acls = null;
                return null;
        }
        return null;
```

Reply:
I thought about this, but IMO it will add minimum value. Let me explain why:

1. When records/unions are created, fields/members are set via the generated setters which do set the ivars. Rarely are the map put methods directly invoked. Even if they are later in the filter chain etc. the object wrapper is gone by then.

2. When decoding the map put is invoked, but there is no record/union wrapper then.